### PR TITLE
Add current_pid function

### LINKED ISSUE #44971: Tracking issue for std::process::id
The standard library provides a way to get the process id of a child process with `std::process::Child::id()`, but doesn't have any way to get the current processes id (the equivalent of `getpid`). 

It is possible to do this with `libc::getpid`, but it seems to me that this should be something that is part of the standard library, especially since it already has cross-platform support for process ids of child processes.

### PR DESCRIPTION
**Pull request by tmccombs**
Fixes #44971

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -1167,6 +1167,24 @@ pub fn abort() -> ! {
...
 }
 
+/// Returns the OS-assigned process identifier associated with this process.
+///
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```no_run
+/// use std::process:current_pid;
```
Comment (cuviper) [inline src/libstd/process.rs]: Missing a colon in `process::current_pid`.

File: src/libstd/sys/redox/os.rs [hunk 2]
```diff
@@ -209,3 +209,7 @@ pub fn exit(code: i32) -> ! {
     let _ = syscall::exit(code as usize);
     unreachable!();
 }
+
+pub fn getpid() -> u32 {
+    syscall::getpid().unwrap() as u32
```
Comment (cuviper) [inline src/libstd/sys/redox/os.rs]: `pid_t` is a `usize` on redox, so this may lose bits.

File: src/libstd/sys/unix/os.rs [hunk 3]
```diff
@@ -513,3 +513,7 @@ pub fn home_dir() -> Option<PathBuf> {
 pub fn exit(code: i32) -> ! {
     unsafe { libc::exit(code as c_int) }
 }
+
+pub fn getpid() -> u32 {
+    unsafe { libc::getpid() as u32 }
```
Comment (cuviper) [inline src/libstd/sys/unix/os.rs]: `pid_t` is technically signed by standard, defined as `i32` on all of `libc`'s unix platforms.

File: src/libstd/process.rs [hunk 4]
```diff
@@ -1167,6 +1167,24 @@ pub fn abort() -> ! {
...
+///
+/// println!("My pid is {}", current_pid());
+/// ```
+///
+///
+#[unstable(feature = "getpid", issue = "44971", reason = "recently added")]
+pub fn current_pid() -> u32 {
```
Comment (cuviper) [inline src/libstd/process.rs]: I think `i64` or `u64` would be more prudent.

Comment (sfackler) [reply src/libstd/process.rs]: We already use u32 elsewhere: https://doc.rust-lang.org/std/process/struct.Child.html#method.id

File: src/libstd/sys/redox/os.rs (hunk 2 above)
Comment (clarcharr) [reply src/libstd/sys/redox/os.rs]: Does Redox really allow PIDs greater than 4 billion-ish though?

Comment (retep998) [reply src/libstd/sys/redox/os.rs]: If it does, we'll need to reconsider the current `Child::id` method.

File: src/libstd/sys/unix/os.rs (hunk 3 above)
Comment (clarcharr) [reply src/libstd/sys/unix/os.rs]: Yes, but `getpid()` is guaranteed to return a non-negative number, and so `u32` is valid here.

File: src/libstd/process.rs (hunk 4 above)
Comment (tmccombs) [reply src/libstd/process.rs]: Yep, that's why I chose u32. Although I think usize or u32 would probably make more sense for both.

File: src/libstd/sys/redox/os.rs (hunk 2 above)
Comment (tmccombs) [reply src/libstd/sys/redox/os.rs]: I made it a u32 to be consistent with `Child::id`.

**Comment by sfackler** [comment]
I think I might lean towards `current_id` rather than `current_pid` since we're already in the `process` module?

**Comment by sfackler** [comment]
@rfcbot fcp merge

**Comment by alexcrichton** [comment]
Looks good to me! I might throw out the names `id` or `current_id`, however to jive better with `Child::id` (we haven't used "pid" yet)

**Comment by carols10cents** [comment]
ping @BurntSushi and @aturon for your ticky boxes [here](https://github.com/rust-lang/rust/pull/45059#issuecomment-334955141)!

**Comment by kennytm** [comment]
Ping @BurntSushi, @aturon again — [ticky boxes](https://github.com/rust-lang/rust/pull/45059#issuecomment-334955141)~

**Comment by aturon** [comment]
Checking off for @BurntSushi as well -- because this is an unstable function, full FCP isn't really needed.

**Pull request merged**